contracts/src/v0.8/automation/dev/chains/ScrollModule.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import {IScrollL1GasPriceOracle} from "../../../vendor/@scroll-tech/contracts/src/L2/predeploys/IScrollL1GasPriceOracle.sol";
+import {IChainModule} from "../interfaces/v2_2/IChainModule.sol";
+
+contract ScrollModule is IChainModule {
+  /// @dev SCROLL_L1_FEE_DATA_PADDING includes 120 bytes for L1 data padding for Optimism
+  /// @dev according to testing, this padding allows automation registry to properly estimates L1 data fee with 3-5% buffer
+  /// @dev this MAY NOT work for a different product and this may get out of date if transmit function is changed
+  bytes internal constant SCROLL_L1_FEE_DATA_PADDING =
+    hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+  /// @dev SCROLL_ORACLE_ADDR is the address of the L1GasPriceOracle precompile on Optimism.
+  /// @dev reference: https://docs.scroll.io/en/developers/transaction-fees-on-scroll/#estimating-the-l1-data-fee
+  address private constant SCROLL_ORACLE_ADDR = 0x5300000000000000000000000000000000000002;
+  IScrollL1GasPriceOracle internal constant SCROLL_ORACLE = IScrollL1GasPriceOracle(SCROLL_ORACLE_ADDR);
+
+  function blockHash(uint256 n) external view returns (bytes32) {
+    if (n >= block.number || block.number - n > 256) {
+      return "";
+    }
+    return blockhash(n);
```

Comment:
we could potentially make this the "default" `blockHash()` function in the Base module, and then scroll, OP, and arbitrum can all use the same inherited function :)

Reply:
done